apps/android/app/src/main/java/chat/simplex/app/model/ChatModel.kt

```diff
@@ -486,6 +486,24 @@ data class Chat (
   val chatItems: List<ChatItem>,
   val chatStats: ChatStats = ChatStats(),
 ) {
+  val userCanSend: Boolean
+    get() = when (chatInfo) {
+      is ChatInfo.Direct -> true
+      is ChatInfo.Group -> {
+        val m = chatInfo.groupInfo.membership
+        m.memberActive && m.memberRole >= GroupMemberRole.Member
+      }
+      else -> false
+    }
+
+  val userIsObserver: Boolean get() = when(chatInfo) {
+    is ChatInfo.Group -> {
+      val m = chatInfo.groupInfo.membership
+      m.memberActive && m.memberRole == GroupMemberRole.Observer
+    }
+    else -> false
+  }
```

Comment:
could userCanSend in case of group be defined as !userIsObserver? these functions seem to almost mirror one another

Reply:
not exactly